src/jit/instr.h

```diff
@@ -280,10 +280,29 @@ enum emitAttr : unsigned
 enum InstructionSet
 {
 #ifdef _TARGET_XARCH_
-    InstructionSet_SSE2,      // SSE2 Instruction set
-    InstructionSet_SSE3_4,    // SSE3, SSSE3, SSE4.1 and SSE4.2 instruction set
-    InstructionSet_AVX,       // AVX2 instruction set
-                              // TODO-Cleaup - This should be named as InstructionSet_AVX2
+    // Linear order start
+    InstructionSet_ILLEGAL = 0,
+    InstructionSet_SSE     = 1,
+    InstructionSet_SSE2    = 2,
+    InstructionSet_SSE3    = 3,
+    InstructionSet_SSSE3   = 4,
+    InstructionSet_SSE41   = 5,
+    InstructionSet_SSE42   = 6,
+    InstructionSet_SSE3_4  = 7,    // SSE3, SSSE3, SSE4.1 and SSE4.2 instruction set
+    InstructionSet_AVX     = 8,
+    InstructionSet_AVX2    = 9,
+    // Linear order end
+    // Instruction sets have the linear order only in above area.
+    // Values of InstructionSet not in this area cannot be compared 
+    // (i.e. compiler->getSIMDInstructionSet() >= InstructionSet_SSE3_4).
```

Comment:
It appears that the values can no longer be compared at all. Indeed, this change would seem to break the existing code that does comparisons, since if any of the higher bits are set, it will be true that `compiler->getSIMDInstructionSet() >= InstructionSet_AVX2`, which could lead to problems. I don't think you can make this change without changing all the code that depends on the values of `compiler->getSIMDInstructionSet()` being in linear order.

Reply:
> It appears that the values can no longer be compared at all.

I agree, and this is a temporary solution. Actually, only SIMD intrinsic (`System.Numerics.Vectors`) code is comparing instruction set values, and the new hardware intrinsic implementation never depends upon the order. In this PR, SIMD still works well because `getSIMDInstructionSet` only returns SSE2, SSE3_4, or AVX. I will refactor this "ISA comparison" in https://github.com/dotnet/coreclr/issues/14065

Reply:
> I agree, and this is a temporary solution.

Before merging this, then, you need to add a comment indicating what the longer term solution is, and how it will interplay with `getFloatingPointInstructionSet()` (and `getSIMDInstructionSet()`).

Reply:
Got it. Will do.